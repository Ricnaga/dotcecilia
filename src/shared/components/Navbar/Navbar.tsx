import { Button } from '../Button/Button';
import { useNavbar } from './hooks/useNavbar';

export function Navbar() {
  const {
    data: { links },
    functions: { logOut },
  } = useNavbar();

  return (
    <div className="flex items-center">
      <nav className="grow">
        {links.map(({ href, children }) => (
          <a
            key={href}
            href={href}
            className="text-xl leading-4 font-medium mr-4 hover:text-sky-950 hover:drop-shadow-lg transition-all"
          >
            {children}
          </a>
        ))}
      </nav>
      <Button className="text-lg" variants="slide" onClick={logOut}>
        Sair
      </Button>
    </div>
  );
}
