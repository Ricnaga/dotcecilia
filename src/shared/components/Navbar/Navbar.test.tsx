import { render } from '@testing-library/react';
import { Navbar } from './Navbar';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useNavigate: jest.fn(),
}));

describe('Component: Navbar', () => {
  it('should render correctly', () => {
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
