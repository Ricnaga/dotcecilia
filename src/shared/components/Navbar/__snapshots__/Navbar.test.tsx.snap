// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Navbar should render correctly 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <nav
      class="grow"
    >
      <a
        class="text-xl leading-4 font-medium mr-4 hover:text-sky-950 hover:drop-shadow-lg transition-all"
        href="calculator"
      >
        Calculadora
      </a>
      <a
        class="text-xl leading-4 font-medium mr-4 hover:text-sky-950 hover:drop-shadow-lg transition-all"
        href="paystub"
      >
        Holerite
      </a>
      <a
        class="text-xl leading-4 font-medium mr-4 hover:text-sky-950 hover:drop-shadow-lg transition-all"
        href="advancepayment"
      >
        Adiantamento
      </a>
      <a
        class="text-xl leading-4 font-medium mr-4 hover:text-sky-950 hover:drop-shadow-lg transition-all"
        href="agreement"
      >
        Acerto
      </a>
    </nav>
    <button
      class="text-lg"
      data-variants="slide"
    >
      <span
        data-text="default"
      >
        Sair
      </span>
    </button>
  </div>
</div>
`;
